src/compiler/emitter.ts

```diff
@@ -731,7 +737,11 @@ namespace ts {
         }
 
         function emitBodyIndirect(node: Node, elements: NodeArray<Node>, emitCallback: (node: Node) => void): void {
-            if (emitBodyWithDetachedComments) {
+            // If the node is a sourceFile and it has prologueDirective (statmentOffSet is not zero) that is synthesize
+            // We will need to emit detached comment here because emitPrologueDirective will not emit comments of such prologue directive
+            const shouldEmitDetachedComment = node.kind !== SyntaxKind.SourceFile ? true :
+                statementOffset === 0 || nodeIsSynthesized((node as SourceFile).statements[statementOffset - 1]);

```

Comment:
You can perform a trivial check for `elements.length === 0 || !isPrologueDirective(elements[0])` in place of `statementOffset === 0` so that we don't have to close over `statementOffset`.